File: lib/190.reverse_bits.dart
void main() {
  print(reverseBits(18432));
  print(reverseBits(43261596));
}

int reverseBits(int n) {
  var result = 0;
  for (var i = 0; i < 32; i++) {
    result <<= 1;
    result += (n >> i) & 1;
  }
  return result;
}

File: src/190.reverse_bits.kt
fun main() {
    println(reverseBits(18432))
    println(reverseBits(43261596))
}

/// todo 無法確認編譯器如何把Long放入
fun reverseBits(n: Int): Int {
    var result = 0
    for (i in 0..31) {
        result = result.shl(1)
        result = result or (n shr i and 1)
    }
    return result
}